packages/drivers/odsp-driver/src/test/epochTestsWithRedemption.spec.ts

```diff
@@ -62,14 +62,13 @@ describe("Tests for Epoch Tracker With Redemption", () => {
 
         try {
             // We will trigger a successful call to return the value set in the cache after the failed joinSession call
-            epochCallback.setCallback(async () => mockFetch({}, async () => {
-                return epochTracker.fetchFromCache(cacheEntry1, undefined, "other");
-            }, okResponse, false, false));
+            epochCallback.setCallback(async () => epochTracker.fetchFromCache(cacheEntry1, undefined, "other"));
 
             // Initial joinSession call will return 404 but after the timeout, the call will be retried and succeed
```

Comment:
@jatgarg, This comment does not match reality. I believe my refactoring preserved one to one how code worked before.
There was no 404 / notFound handling as far as I can see from debugging it before changes.
Are we missing a test case here?

Reply:
@sumedhb1995 Can you take a look here as you added this test case here?

Reply:
The initial call to set the value in the cache should succeed (i.e. the old lines 65-67) with the Ok response parameter but the subsequent joinSession call (i.e. the old line 70-72) should fail with the notFound response parameter

Reply:
So this is where we were seeing the 404 show up within the epochTracker's fetchAndParseAsJSON call
![image](https://user-images.githubusercontent.com/7992711/109723727-59975200-7b63-11eb-92e0-3ad5d84b0674.png)
Since fetchType is "joinSession", it will automatically try again and the subsequent call succeeds. That's why it never enters the catch block on the test and we assert at the end on "success" being true

Reply:
I believe we can add this condition back with the following change for the joinSession call
```javascript
 await mockFetchMultiple(
                [notFound(), okResponse({ "x-fluid-epoch": "epoch1" }, {})],
                async () => epochTracker.fetchAndParseAsJSON("fetchUrl", {}, "joinSession"),
            );
```

Reply:
Yes, you are right, I did not got all the details on how it was working before. Copied code :)